### PR TITLE
chore(deps): bump linters & update lint rules

### PR DESCRIPTION
**Pull request by yoshinorin**
## Purpose

Current version of `@typescript-eslint` does not support `typescript 4.6.3`.
This PR purpose is update linters (`eslint` & `typescript-eslint`)

```sh
$ yarn run lint
yarn run v1.22.11
warning metals@1.13.10: The engine "vscode" appears to be invalid.
$ eslint . --ext .ts --fix && yarn format
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.6.3

Please only submit bug reports when using the officially supported version.

=============
```

Thank you :)

## Others

I checked the `eslint v8.0.0` migration guide and the `typescript-eslint 5.0.0` release note. One of the breaking change found but basically seems no problem.

- https://eslint.org/docs/8.0.0/user-guide/migrating-to-8.0.0
- https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ module.exports = {
           "warn",
           { ignoreRestArgs: true },
         ],
+        "no-unused-vars": "off",
+        "@typescript-eslint/no-unused-vars": ["warn"],
```
Comment (yoshinorin) [inline .eslintrc.js:20]: Breaking change in `typescript-eslint v5.0.0`.

> https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0
> https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unused-vars.md

**Comment by yoshinorin** [comment]
Opps.. I'm going to resolve conflict.

**Comment by kpodsiad** [comment]
@yoshinorin could you fix eslint CI too? It should fail when there are some warnings but currently it doesn't. I guess it's a matter of some flag.

**Comment by yoshinorin** [comment]
@kpodsiad 

> could you fix eslint CI too? It should fail when there are some warnings but currently it doesn't. I guess it's a matter of some flag.

OK. I will try 👌

File: src/scalaDebugger.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ class ScalaMainConfigProvider implements vscode.DebugConfigurationProvider {
         path: editor.document.uri.toString(true),
         runType: RunType.RunOrTestFile,
       };
-      return start(false, args).then((_) => {
+      return start(false, args).then(() => {
```
Comment (yoshinorin) [inline src/scalaDebugger.ts:78]: delete unused var

File: src/util.ts [hunk 3]
```diff
@@ -25,7 +25,9 @@ export function getConfigValue<A>(
   key: string
 ): { value: A; target: ConfigurationTarget } | undefined {
   const value = config.get<A>(key);
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
   const { defaultValue, workspaceValue } = config.inspect<A>(key)!;
+  /* eslint-enable @typescript-eslint/no-non-null-assertion */
```
Comment (yoshinorin) [inline src/util.ts:30]: I add `eslint-disable @typescript-eslint/no-non-null-assertion` for `non-null-assertion`.

**Comment by yoshinorin** [comment]
Done :)
I updated `eslint` rules.

**Review by kpodsiad — APPROVED** [review]
Looks good! Thanks @yoshinorin!
Are you on Scalameta Discord? I wanted to ask you one question, but it'll be easier on DM or some chat rather than Github discussions.

**Comment by yoshinorin** [comment]
@kpodsiad 

> Are you on Scalameta Discord?

I joined Scalameta Discord just now :)

**Pull request merged**